src/main/views/viewManager.ts

```diff
@@ -112,44 +121,76 @@ export class ViewManager {
         this.configServers.forEach((server) => this.loadServer(server));
     }
 
+    /** Called when a new configuration is received
+     * Servers or tabs have been added or edited. We need to
+     * close, open, or reload tabs, taking care to reuse tabs and
+     * preserve focus on the currently selected tab. */
     reloadConfiguration = (configServers: TeamWithTabs[]) => {
-        this.configServers = configServers.concat();
-        const oldviews = this.views;
+        const focusedTuple: TabTuple | undefined = this.views.get(this.currentView as string)?.urlTypeTuple;
+
+        const current: Map<TabTuple, MattermostView> = new Map();
+        for (const x of this.views.values()) {
+            current.set(x.urlTypeTuple, x);
+        }
+
+        const views: Map<TabTuple, MattermostView> = new Map();
+        const closed: Map<TabTuple, {srv: MattermostServer; tab: Tab; name: string}> = new Map();
+
+        const sortedTabs = configServers.flatMap((x) =>
```

Comment:
I'm having a tough time figuring out what we're trying to do here. This is making an array of tuples for each server's tabs, is that right? It seems a bit odd that we'd be both combining them into a single array (presumably for easier iteration) but then sorting them when they won't be displayed that way.

Reply:
Yes, that's right. I'm sorting them because the original implementation did as well, at line 119. I am not sure if it serves any purpose, so I elected to trust the original implementation had something in mind that I'm not aware of.